### ISSUE TITLE
Difficulties in setting up DPCTF Test Runner for testing on a TV

### ISSUE THREAD

**Issue by yanj-github** [cta-wave/dpctf-deploy#2]
For the WMAS test runner we can test for this architecture on a DUT as we point to the instance in the cloud on the public network.

For the DPCTF test runner we are having difficulties setting this up with the server locally due to the absence of a DNS server, where the suggested path was to use a router and configure a local domain on the router. Our company IT have pushed back on this approach and we believe others may have similar issues. 

As a first step, could we please have clearly documented setup steps and a diagram confirming the correct way to run the DPCTF test runner in case our understanding is wrong. Currently we appear to be blocked by needing to setup a DNS server at the server end of the solution which is not a trivial task.

**Comment by louaybassbouss** [comment]
@yanj-github can you confirm that the following setup works on your side:

1. Install DPCTF Test Runner on a computer `A` e.g. with IP `192.168.1.2`
2. On a second computer `B` in local network, update the hosts file and add a new entry for `web-platform.test` pointing to the IP of the Test Runner e.g. `192.168.1.2       web-platform.test`
3. Open the Landing page http://web-platform.test:8000/_wave/index.html in a browser on computer `B` and check if it works.

If this works then all you need is to configure a local domain pointing to the IP of the test runner and use this new domain in the config if it is different from the default domain `web-platform.test`. All network configurations are not part of the project.

What diagram you need?

**Comment by yanj-github** [comment]
Thanks @louaybassbouss I can access Test Runner runing on computer A from computer B by updating the host file.
However whe I want to connect the Test Runner from A TV, I cannot resolve the domain as there are not any hosts file on the TV.

**Comment by louaybassbouss** [comment]
> Thanks @louaybassbouss I can access Test Runner runing on computer A from computer B by updating the host file.
> However what I want to connect the Test Runner from A TV, I cannot resolve the domain as there are not any hosts file on the TV.

Yes just to make sure that there are no other issues in the local connection but everything looks fine.

**Comment by yanj-github** [comment]
> 
> 
> > Thanks @louaybassbouss I can access Test Runner runing on computer A from computer B by updating the host file.
> > However what I want to connect the Test Runner from A TV, I cannot resolve the domain as there are not any hosts file on the TV.
> 
> Yes just to make sure that there are no other issues in the local connection but everything looks fine.

Yes @louaybassbouss  to host the server on a different device than the DUT by updating a host file is fine. Updating host file to resoving domain to IP is not a problem on a PC. However, for DUTs such as TV where there is no hosts file, can I have a clear instrcution of what it should be done to resolve the domain please? Thank you!

**Comment by louaybassbouss** [comment]
> > > Thanks @louaybassbouss I can access Test Runner runing on computer A from computer B by updating the host file.
> > > However what I want to connect the Test Runner from A TV, I cannot resolve the domain as there are not any hosts file on the TV.
> > 
> > 
> > Yes just to make sure that there are no other issues in the local connection but everything looks fine.
> 
> Yes @louaybassbouss to host the server on a different device than the DUT by updating a host file is fine. Updating host file to resoving domain to IP is not a problem on a PC. However, for DUTs such as TV where there is no hosts file, can I have a clear instrcution of what it should be done to resolve the domain please? Thank you!

configure your router with a local domain name that points to the IP of the test runner.

**Comment by yanj-github** [comment]
If I were in office this is fine, however, this is not something that I can easily set up in home environment. The router that I have does not support this feature and now I have difficulties in finding a router that it supports this. It would be highly appreciated that if you can recommend a router if you know that it supports this feature please?

**Comment by louaybassbouss** [comment]
@yanj-github we added a new config parameter [host_override](https://github.com/cta-wave/dpctf-test-runner/blob/master/tools/wave/docs/config.md#211-host-override) where you can specify the IP address of the Test Runner. You need to add this parameter in the [config.json](https://github.com/cta-wave/dpctf-deploy/blob/master/config.json) file. This way, you can directly use the IP address in the url instead of domain name. Example: http://192.168.1.2:8000/_wave/index.html if 192.168.1.2 is the IP of the Test Runner. In this case you should add `"host_override": "192.168.1.2"` to the config.json file. 

PS: this feature is not supported by the WPT or WMAS Test Runner and only added to the DPCTF Test Runner.

**Comment by FritzHeiden** [comment]
@yanj-github The `host_override` parameter is part of the [new release](https://github.com/cta-wave/dpctf-test-runner/releases/tag/v1.0.0). Please try setting it to your host machines IP and report if this solve your issue

**Comment by yanj-github** [comment]
Sorry for the delay in response. I just have managed to try this out. Thank you very much @louaybassbouss  and @FritzHeiden  for getting this implemented. It is working.
